src/database.h

```diff
@@ -161,11 +159,8 @@ class DBResult
 			}
 
 			T data;
-			try {
-				data = boost::lexical_cast<T>(row[it->second]);
-			} catch (boost::bad_lexical_cast&) {
-				data = 0;
-			}
+			std::stringstream ss(row[it->second]);
+			ss >> data;
```

Comment:
You are right, according to my tests string stream is generally 3x slower than `boost::lexical_cast` except for float.

Specialized templates works best and is 2x faster than `boost::lexical_cast` on my machine, but a whopping 10x faster for floats.

Sample code: https://gist.github.com/ranisalt/7bb2af8b7fa21dd7eeb2b4e8f7aef156
I will create a cast function that both `pugi::cast` and `DBResult` will use based on specialized templates.